VoodooI2C/VoodooI2C/VoodooI2CDevice/VoodooI2CDeviceNub.cpp

```diff
@@ -193,17 +201,21 @@ IOReturn VoodooI2CDeviceNub::getDeviceResources() {
         return kIOReturnNotFound;
     }
 
-    if (crs_parser.found_gpio_interrupts ||
+    if (!force_polling) {
+        if (crs_parser.found_gpio_interrupts ||
         (!validateInterrupt() && getAlternativeInterrupt(&crs_parser) == kIOReturnSuccess)) {
         setProperty("gpioPin", crs_parser.gpio_interrupts.pin_number, 16);
         setProperty("gpioIRQ", crs_parser.gpio_interrupts.irq_type, 16);
 
         has_gpio_interrupts = true;
         gpio_pin = crs_parser.gpio_interrupts.pin_number;
         gpio_irq = crs_parser.gpio_interrupts.irq_type;
+        }
+    } else {
+        IOLog("%s::%s Forced polling mode, skipping APIC/GPIO interrupts", controller_name, getName());
```

Comment:
How about moving the prompt above and reduce one more parentheses?

Reply:
And missing a `\n` here, I also missed one two lines below lol.

Reply:
> And missing a `\n` here, I also missed one two lines below lol.

Is `\n` really required? I've never used it with `IOLog` nor `os_log` (DriverKit).

Reply:
It will be printed in separate lines with -v. And with Big Sir, kext injected from bootloader can not output to system log. Then it's necessary to grep keywords from `dmesg`.

Reply:
It will be printed in separate lines with -v. And with Big Sir, kext injected from bootloader can not output to system log. Then it's necessary to grep keywords from dmesg.

Reply:
It still is printed on separate lines (I checked). I omitted `\n` as the statement looks a bit cleaner. Also, in any of Acidanthera kexts, I don't see them using `\n` either.

Reply:
They baked it into the macro: https://github.com/acidanthera/Lilu/blob/master/Lilu/Headers/kern_util.hpp#L79

Reply:
And some bad example via `sudo dmesg | grep 'Sandbox'`:
```
[ 8571.368915]: Sandbox: com.apple.Safari(455) deny(1) user-preference-read kCFPreferencesAnyApplication[IGFB][LOG  ][TRANSACTIO] [Flip_Transaction] FB0: Flipping display with wsaa 1 flipped from 3840 x 2160 -> 3840 x 2160 on 4234c000 with 801 options at stride 15360
```

Reply:
Alright. Got it.